lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2466,6 +2466,38 @@ struct TritonGPUInferLayoutInterface
                                              Attribute srcEnc,
                                              ArrayRef<int64_t> dstShape,
                                              Attribute &dstEnc) const {
+    if (auto mmaEncoding = dyn_cast<NVMMASharedEncodingAttr>(srcEnc)) {
```

Comment:
This is codemotion from OptimizeDotOperands, by noting that inferring the src for reshape and inferring the dst is the same